src/info.cc

```diff
@@ -240,6 +240,7 @@ void printInfo(Zypper & zypper, const std::vector<std::string> &names_r, const P
 
     if ( q.empty() )
     {
+      zypper.setExitInfoCode( ZYPPER_EXIT_INF_CAP_NOT_FOUND );
```

Comment:
I'd set the ExitInfo a bit later, if the fallback query is empty or no fallback is wanted (the two `continue` cases).

Reply:
Good thinking, I might've tried that but I'll try moving the `setExitInfoCode` call again.